pkg/clusteragent/admission/mutate/auto_instrumentation_test.go

```diff
@@ -79,11 +93,28 @@ func TestInjectAutoInstruConfig(t *testing.T) {
 			expectedEnvVal: "/datadog-lib/:predefined",
 			wantErr:        false,
 		},
+		{
+			name:    "PYTHONPATH set via ValueFrom",
+			pod:     fakePodWithEnvFieldRefValue("python-pod", "PYTHONPATH", "path"),
+			lang:    "python",
+			image:   "gcr.io/datadoghq/dd-lib-python-init:v1",
+			wantErr: true,
+		},
+		{
+			name:    "Unknown language",
+			pod:     fakePod("unknown-pod"),
+			lang:    "unknown",
+			image:   "gcr.io/datadoghq/dd-lib-unknown-init:v1",
+			wantErr: true,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			err := injectAutoInstruConfig(tt.pod, tt.lang, tt.image)
-			require.False(t, (err != nil) != tt.wantErr)
+			if err != nil {
+				require.True(t, tt.wantErr)
+				return
+			}
```

Comment:
If `err` is `nil`, the value of `tt.wantErr` will not be checked so that, if the test case is supposed to return an error (`tt.wantErr = true`), but doesn’t (`err = nil`), it will not be caught.

Reply:
thanks @L3n41c ! fixed it